.github/workflows/nightly_build.yaml

```diff
@@ -4,16 +4,18 @@ on:
     # Random minute number to avoid GH scheduler stampede
     - cron: '37 21 * * *'
   workflow_dispatch: {}
-permissions:
-  contents: read
-  packages: write
 
 env:
   NIGHTLY: true
 
 jobs:
   build-and-publish-images:
     runs-on: ubuntu-20.04
+
+    permissions:
```

Comment:
Moved to job level, so we don't run the chance new job added in this workflow will also have write access by default.